upload-and-download

IlyaKirdun opened the pull request:
Добавил файлы uploadAndDownload.page ; uploadAndDownload.spec(2 кейса) 
Добавил папку testFiles с вложенным файлом fileForUpload

2 теста для страницы Upload and Download:

Проверяем функционал кнопки "Download";
Проверяем функционал кнопки "Выберите файл".

IlyaKirdun committed:
Создал файл uploadAndDownload.page
Создал папку testFiles и добавил fileForUpload

IlyaKirdun committed:
Добавил метод deleteDownloadFile() в uploadAndDownload.page
Создал файл uploadAndDownload.spec

rybignat commented on page/uploadAndDownload.page.ts:
```diff
@@ -0,0 +1,45 @@
+    await this.downloadButton.click()
+    const download = await downloadPromise
+
+    await download.saveAs('utils/downloads/' + download.suggestedFilename())
```
А почему файл скачивается по кастомному пути,если есть путь по умолчанию?

IlyaKirdun committed:
Изменил переменную downloadFilePath.
Переписал метод clickDownloadButton().

IlyaKirdun committed:
Добавил переменная в .env.example;
Переместил папку testFiles в корневую директорию;
Добавил методы: deleteFile(), verifyFileExist() в файл functions;
Переделал методы в uploadAndDownload.page: clickDownloadButton(); clickUploadFileButtonAndSelectFile()
Добавил hook в uploadAndDownload.spec

IlyaKirdun committed:
Добавил skip теста - Проверяем функционал кнопки "Download". для GITHUB_ACTIONS

IlyaKirdun committed:
smallFix
